src/arviz_stats/psens.py

```diff
@@ -0,0 +1,213 @@
+        Any object that can be converted to an :class:`arviz.InferenceData` object.
+        Refer to documentation of :func:`arviz.convert_to_dataset` for details.
+        For ndarray: shape = (chain, draw).
+        For n-dimensional ndarray transform first to dataset with ``az.convert_to_dataset``.
+    group : {"log_prior", "log_likelihood"}, default "log_prior"
+        When `component` is "likelihood", the log likelihood values are retrieved
+        from the ``log_likelihood`` group as pointwise log likelihood and added
+        together. With "prior", the log prior values are retrieved from the
+        ``log_prior`` group.
+    component_var_names : str, optional
+        Name of the prior or log likelihood variables to use
+    component_coords : dict, optional
+        Coordinates defining a subset over the component element for which to
+        compute the prior sensitivity diagnostic.
+    var_names : list of str, optional
+        Names of posterior variables to include in the power scaling sensitivity diagnostic
+    coords : dict, optional
+        Coordinates defining a subset over the posterior. Only these variables will
+        be used when computing the prior sensitivity.
+    filter_vars: {None, "like", "regex"}, default None
+        If ``None`` (default), interpret var_names as the real variables names.
+        If "like", interpret var_names as substrings of the real variables names.
+        If "regex", interpret var_names as regular expressions on the real variables names.
+    delta : float
+        Value for finite difference derivative calculation.
+
+
+    Returns
+    -------
+    xarray.DataTree
+        Returns dataTree of power-scaling sensitivity diagnostic values.
+        Higher sensitivity values indicate greater sensitivity.
+        Prior sensitivity above 0.05 indicates informative prior.
+        Likelihood sensitivity below 0.05 indicates weak or non-informative likelihood.
+
+    Notes
+    -----
+    The diagnostic is computed by power-scaling the specified component (prior or likelihood)
+    and determining the degree to which the posterior changes as described in [1]_.
+    It uses Pareto-smoothed importance sampling to avoid refitting the model.
+
+    References
+    ----------
+    .. [1] Kallioinen et al, *Detecting and diagnosing prior and likelihood sensitivity with
+       power-scaling*, 2022, https://arxiv.org/abs/2107.14054
+    """
+    dataset = extract(dt, var_names=var_names, filter_vars=filter_vars, group="posterior")
+    if coords is None:
+        dataset = dataset.sel(coords)
+
+    if group == "log_likelihood":
+        component_draws = get_log_likelihood(dt, var_name=component_var_names, single_var=False)
+    elif group == "log_prior":
+        component_draws = get_log_prior(dt, var_names=component_var_names)
+    else:
+        raise ValueError("Value for `group` argument not recognized")
+
+    component_draws = component_draws.stack(__sample__=("chain", "draw"))
```

Comment:
```suggestion
    component_draws = component_draws.stack(sample=("chain", "draw"))
```

No more `__sample__` in arviz-stats. `az.extract` already uses `sample`, and we have also extended the "reserved" dimensions across arviz-base and arviz-plots. We just need to be sure to document which dimension names are assumed to have a specific meaning in arviz.